pc/PlatformDeviceControl.c

```diff
@@ -266,12 +273,31 @@ libusb_device_handle *usbLinkOpen(const char *path)
     }
     usb_free_device(dev);
 #else
+    struct libusb_device_descriptor desc;
+    int res;
+    if ((res = libusb_get_device_descriptor(dev, &desc)) < 0) {
+                printf("Unable to get USB device descriptor: %s\n", libusb_strerror(res));
+        }
+
+    usb_speed_enum = libusb_get_device_speed(dev);
+    char *speed_str[] = {"Unknown", "Low/1.5Mbps", "Full/12Mbps", "High/480Mbps", "Super/5000Mbps"};
```

Comment:
Last iteration of libusbs speed is SUPER_PLUS = 5. Indexing speed_str[5] is undefined behaviour.
```
libusb_speed {
  LIBUSB_SPEED_UNKNOWN = 0, LIBUSB_SPEED_LOW = 1, LIBUSB_SPEED_FULL = 2, LIBUSB_SPEED_HIGH = 3,
  LIBUSB_SPEED_SUPER = 4, LIBUSB_SPEED_SUPER_PLUS = 5
}
```

Do:
```const char* speed_str[] = ...``` <- as literals are const
and before mvLog
```
const char* speed = speed_str[0];
if(usb_speed_enum >= 0 && usb_speed_enum < sizeof(speed_str)){
    speed = speed_str[usb_speed_enum];
}
```